src/App.tsx

```diff
@@ -1,10 +1,64 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { ErrorInfo } from './components/Errorinfo';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorType } from './types/ErrorType';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 12017;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<FilterType>(
+    FilterType.All,
+  );
+  const [errorMsg, setErrorMsg] = useState<ErrorType | null>(null);
+
+  const errorFound = (error: ErrorType) => {
+    setErrorMsg(error);
+    setTimeout(() => setErrorMsg(null), 3000);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodosFromServer)
+      .catch(() => errorFound(ErrorType.TodosNotLoaded));
+  }, []);
+
+  const filterTodos = useCallback(
+    (todos: Todo[]) => {
+      switch (selectedFilter) {
+        case FilterType.All:
+          return todos;
+
+        case FilterType.Active:
+          return todos.filter(todo => !todo.completed);
+
+        case FilterType.Completed:
+          return todos.filter(todo => todo.completed);
+
+        default:
+          return todos;
+      }
+    },
```

Comment:
It would be good to move the filtering logic to a separate handler to improve code readability